src/server/saved_objects/service/lib/repository.js

```diff
@@ -464,6 +464,69 @@ export class SavedObjectsRepository {
     };
   }
 
+  /**
+   * Increases a counter field by one. Creates the document if one doesn't exist for the given id.
+   *
+   * @param {string} type
+   * @param {string} id
+   * @param {string} counterFieldName
+   * @param {object} [options={}]
+   * @property {object} [options.migrationVersion=undefined]
+   * @returns {promise}
+   */
+  async incrementCounter(type, id, counterFieldName, options = {}) {
+    const {
+      migrationVersion,
+    } = options;
+
+    const time = this._getCurrentTime();
+
+
+    const migrated = this._migrator.migrateDocument({
+      id,
+      type,
+      attributes: { [counterFieldName]: 1 },
+      migrationVersion,
+      updated_at: time,
+    });
+
+    const raw = this._serializer.savedObjectToRaw(migrated);
+
+    const response = await this._writeToCluster('update', {
+      id: this._serializer.generateRawId(undefined, type, id),
```

Comment:
@kobelb I saw you just recently made some changes to serializers for namespaces, I'd appreciate it if you could look at how I've used them here. I don't think we need to pass a namespace in here since it's for telemetry but I may be wrong.

Reply:
If we want this method to work with saved object types that are namespaced, which is almost all of them, we'll want to pass in the `namespace` in the options and use it here. When a saved object type is namespaced and a namespace is specified with the options, the raw document `id` in elasticsearch becomes: `${namespace}:${type}:${user_facing_id}`.

Reply:
ah that's a good point, I was thinking too narrowly here